### PR TITLE
feat(gridList): adds grid-based layout

### LINKED ISSUE #933: mdGridList:  Implement new md-gridlist component
#### Overview 

A grid list is best suited to presenting a homogenous data type, typically images, and is optimized for visual comprehension and differentiating between like data types.

See the Material Design Specification for Details:  [Gridlist Specs](https://www.google.com/design/spec/components/grids.html#grids-usage)

The developer will need to implement an Angular Material version that conforms to the functionality and UX defined in the Design specification [above]. 

> Developers should use existing Angular Material components and coding patterns as guides for their own implementation. Shell component assets are available for quick starts [see Below].

#### Important Considerations

*  Consider a `md-gridlist-item` to support needs of tile header and footers, plan to support the **primary** and **secondary** actions specified in the Design docs.
  *  Use css to define the dimensions of each md-gridlist-item.
*  Consider need for `md-image` for image tiles
*  Consider image background loading, promise-based load indicator, image-loaded caching, and transforms for scaling; may be part of functionality associated within an `md-image` component.

```html
<md-gridlist>
   <md-gridlist-item>
     <!-- 
            This is a tile that supports ideas of headers and footers;
            where images have precedence over text.
      -->
     <md-image />
  </md-gridlist-item>
</md-gridlist>
```

*  Static and dynamic (using ng-repeat) usages should be considered.
*  Questions of ripple inks and activity indicators should be considered.

> The component implementation may be dependent upon the `md-icon` rewrite and availability.

### PR DESCRIPTION
**Pull request by shyndman**
Still needs:
* Height calculation (currently grids have a height of 0px)
* Additional styles for header/footer
* Styles for md-button
* Tests

Fixes #933

### CODE REVIEW

**Comment by shyndman** [comment]
I'm the author of all commits.

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "gulp-ng-html2js": "^0.1.8",
     "gulp-plumber": "^0.6.6",
     "gulp-rename": "^1.2.0",
-    "gulp-sass": "ajoslin/gulp-sass#master",
+    "gulp-sass": "^1.3.2",
```
Comment (shyndman) [inline package.json]: We can make this a separate PR if you want, but it seems to be working alright.

**Comment by shyndman** [comment]
Not exactly sure how the PR got into this state. History rewriting?

**Comment by shyndman** [comment]
OK, all fixed up.

**Comment by ThomasBurleson** [comment]
@shyndman - what do you mean the height == 0px ?

**Comment by ThomasBurleson** [comment]
@shyndman - those demos are "slicker than goose poop on a hot metal roof". #woot.

**Comment by ThomasBurleson** [comment]
@shyndman - I will merge and close this. Meanhwhile can you do **another** PR with jsDocs for the MGridTileFooter and MGridTileHeader.

**Comment by shyndman** [comment]
@ThomasBurleson re: height == 0px

Because the grid contains only absolutely positioned elements it doesn't have have a height. I need to write some code to give it one based on the content, but I'm still figuring out the right place to plug in.

**Comment by shyndman** [comment]
@ThomasBurleson re: header and footer

They're described in the `md-grid-tile` docs. Think that's enough?

**Comment by ThomasBurleson** [comment]
@shyndman - perhaps at the end of of the `layout()` function?

**Comment by shyndman** [comment]
@ThomasBurleson It needs access to the element and the rowCount to make it happen, so I think it might have to be callback provided to $mdLayout, similar to map().

**Comment by ThomasBurleson** [comment]
Capture those in a bind curry wrapper

On Monday, February 9, 2015, Scott Hyndman <notifications@github.com> wrote:

> @ThomasBurleson <https://github.com/ThomasBurleson> It needs access to
> the element and the rowCount to make it happen, so I think it might have to
> be callback provided to $mdLayout, similar to map().
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/material/pull/1433#issuecomment-73610614>.
>


-- 
*Thomas Burleson*
Mindspace LLC

http://www.gridlinked.info
http://www.linkedin/in/ThomasBurleson
https://github.com/ThomasBurleson

**Comment by turkingj** [comment]
+1 for height calculation!!

**Comment by shyndman** [comment]
@turkingj I'll see if I can get it in today.

**Comment by turkingj** [comment]
thanks @shyndman, awesome work, much appreciated...
